### DISCUSSION TITLE
关于日志输出请教

### DISCUSSION THREAD

**Discussion by ericzhanchina** [cloudwu/skynet#1734] [General]
之前我是用lua写了一个logger服务，负责写日志到磁盘，并且按照小时分割文件，最近不知道为啥，经常出现某个服务发给日志服务死循环的错误。于是我就改用skynet.error来输出日志了，使用logrotate轮换日志。logrotate的配置文件如下
```

    size 100K
    rotate 24
    missingok
    notifempty
    create 644 root root
    compress
    delaycompress
    dateext
    dateformat -%Y%m%d-%s
```

config.xxx配置文件如下：
```
root        =   "./"
skynet      =   root.."skynet/"
thread      =   1                                                   -- worker thread equals cpus            
logger    =   root.."/log/login.log"
--logger      =   "logger"
logservice  =   "logger"
logpath     =   "./log/"
loggroup    =   "login"
```
现在发现，logrotate把文件轮转分割了之后，skynet还是会向这个改名的文件进行追加，并不会往新创建的比如login.log写入，这个怎么处理能让他往新创建的文件写入呢？

**Comment by cloudwu** [comment]
skynet 自带的 logger 服务有处理 HUP 信号（通过消息 PTYPE_SYSTEM ）你要确保 logrotate 正确发送了该信号。

https://github.com/cloudwu/skynet/blob/master/skynet-src/skynet_start.c#L124

**Reply by ericzhanchina** [reply]
感谢，已经处理好了。
